File: Rio/Rio.Web/Modules/ServerTypes/Workspace/ExamSectionForm.ts
import { LookupEditor, StringEditor, TextAreaEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface ExamSectionForm {
    ExamId: LookupEditor;
    Name: StringEditor;
    Description: TextAreaEditor;
    ParentId: LookupEditor;
}

export class ExamSectionForm extends PrefixedContext {
    static formKey = 'Workspace.ExamSection';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!ExamSectionForm.init)  {
            ExamSectionForm.init = true;

            var w0 = LookupEditor;
            var w1 = StringEditor;
            var w2 = TextAreaEditor;

            initFormType(ExamSectionForm, [
                'ExamId', w0,
                'Name', w1,
                'Description', w2,
                'ParentId', w0
            ]);
        }
    }
}

File: Rio/Rio.Web/Modules/Workspace/ExamSection/ExamSectionForm.cs
using Serenity.ComponentModel;
using Serenity.Web;
using System;

namespace Rio.Workspace.Forms
{
    [FormScript("Workspace.ExamSection")]
    [BasedOnRow(typeof(ExamSectionRow), CheckNames = true)]
    public class ExamSectionForm
    {
        [HalfWidth]
        public string Name { get; set; }
        [TextAreaEditor(Rows = 5)]
        [HalfWidth]
        public string Description { get; set; }
        [HalfWidth]
        public long ExamId { get; set; }
        [HalfWidth]
        public int ParentId { get; set; }
        
    }
}